samples/softsim_external_profile/src/main.c

```diff
@@ -157,16 +170,21 @@ void serial_cb(const struct device *dev, void *user_data) {
   }
 
   while (uart_irq_rx_ready(uart_dev)) {
-    *rx_buf_pos += uart_fifo_read(uart_dev, &rx_buf[*rx_buf_pos], rx_buf_len - *rx_buf_pos);
+    rx_recv = uart_fifo_read(uart_dev, &rx_buf[*rx_buf_pos], rx_buf_len - *rx_buf_pos);
```

Comment:
```suggestion
    rx_recv = uart_fifo_read(uart_dev, &rx_buf[*rx_buf_pos], 1);
```
Can we end up in a case where `[... data, data, \r, \n]` is received in one `fifo_read` and the length ends up being off by one? 

It is a bit hacky to read one by one -> but we can return earlier